core/trino-main/src/main/java/io/trino/server/ServerMainModule.java

```diff
@@ -427,7 +452,8 @@ protected void setup(Binder binder)
         binder.bind(NodeVersion.class).toInstance(new NodeVersion(nodeVersion));
         discoveryBinder(binder).bindHttpAnnouncement("trino")
                 .addProperty("node_version", nodeVersion)
-                .addProperty("coordinator", String.valueOf(serverConfig.isCoordinator()));
+                .addProperty("coordinator", String.valueOf(serverConfig.isCoordinator()))
+                .addProperty("allocatable_memory", String.valueOf(NodeMemoryConfig.AVAILABLE_HEAP_MEMORY - nodeMemoryConfig.getHeapHeadroom().toBytes()));
```

Comment:
It makes sense. I quickly skimmed and not sure if there is an easy way to get pool size here (in Guice module). Or is there?

Reply:
I didn't reallize the memory pool is also configured in a similar way: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/memory/LocalMemoryManager.java#L60

Maybe it makes sense to bind the memory pool size with Guice and use the bound value in those two places?

Reply:
It would be great but I did not figure out how to do that. How to inject anything to `discoveryBinder(binder).bindHttpAnnouncement("trino")...`.

So far I went with:
```java
        // This replicates logic for computing general pool size from LocalMemoryManager
        // TODO: figure out if it is possible to use Guice to bind common value in LocalMemoryManager and to discoveryBinder
        long generalMemoryPoolSize = Runtime.getRuntime().maxMemory() - nodeMemoryConfig.getHeapHeadroom().toBytes();
        if (!nodeMemoryConfig.isReservedPoolDisabled()) {
            generalMemoryPoolSize -= nodeMemoryConfig.getMaxQueryMemoryPerNode().toBytes();
        }

        // version and announcement
        binder.bind(NodeVersion.class).toInstance(new NodeVersion(nodeVersion));
        discoveryBinder(binder).bindHttpAnnouncement("trino")
                .addProperty("node_version", nodeVersion)
                .addProperty("coordinator", String.valueOf(serverConfig.isCoordinator()))
                .addProperty("memory_pool_size", String.valueOf(generalMemoryPoolSize));
```